docs/source/notes/autograd.rst

```diff
@@ -58,6 +58,100 @@ will also require them.
     # Optimize only the classifier
     optimizer = optim.SGD(model.fc.parameters(), lr=1e-2, momentum=0.9)
 
+
+.. _locally-disable-grad-doc:
+
+Locally disabling gradient computation
+--------------------------------------
+
+There are two ways to locally disable gradients from Python: no-grad mode and
+inference mode. Below, we compare no-grad mode and inference mode. We also
+compare several other useful mechanisms that may be conflated with the two.
+
+Grad Mode (Default)
+^^^^^^^^^^^^^^^^^^^
+
+Computations are performed in grad mode by default. In grad mode computations
+on tensors that require grad are tracked by autograd so that gradients can be
+computed with respect to the inputs when running ``output.backward()``. Note: this
+is the only mode where requires_grad is relevant. ``requires_grad`` is ignored in
+all other modes.
+
+Setting requires_grad
+^^^^^^^^^^^^^^^^^^^^^
+
+As described `above <excluding-requires_grad_>`_,
+use ``.requires_grad`` for more fine-grained control over which parameters should
+be updated on the backward pass. Parameters with ``requires_grad=True`` will have
+gradients accumulated into their ``.grad`` fields on ``.backward()``.
+
+Whether a tensor ``requires_grad`` also determines whether the computations are
+tracked by autograd in the first place. An operation is only tracked by
+autograd if at least one of its input tensors require grad.
+
+For example, if you need to freeze the bottom part of your model while
+fine-tuning, you can apply ``requires_grad=False`` on layers that you’d like to
+avoid.
+
+Note: you can only set the requires_grad of leaf nodes (i.e., parameters and
+other tensors that don’t have ``grad_fn``). Non-leaf tensor have ``requires_grad=True``
+by definition.
+
+No-grad Mode
+^^^^^^^^^^^^
+
+Computations in no-grad mode behave as if none of the inputs require grad.
+Enable no-grad mode when you need to perform operations that should not be
+tracked by autograd, but you’d still like to use the outputs of these
+computations in grad mode later. If you know that the outputs won’t be involved
+in computations in grad mode later, consider using inference mode.
+
+For example, you would use no-grad if you are writing an optimizer, you’d like to update parameters
+in-place in order to perform a training step without the update being tracked
+by autograd. You also intend to use the updated parameters for computations in
+grad mode in the next forward pass.
+
+Inference Mode
+^^^^^^^^^^^^^^
+
+Inference mode (PyTorch >= 1.9) provides improved performance during the model
```

Comment:
How is this:
```
Inference mode is an extreme version of no-grad mode. Just like in no-grad
mode, computations in inference mode are not tracked by autograd, but
enabling inference mode will allow PyTorch to speed up your model even more.
This better runtime comes with a drawback: tensors created in inference mode
will not be able to be used in computations tracked by autograd after
exiting inference mode.
```
I think its better to be a little more vague here about what exactly the restriction is here

Reply:
Looks better.
I would re-use the wording from above to describe the no grad mode: either both say that requires_grad is ignored or both say that computatioins are not tracked.

Reply:
Ahh I'm a little torn between the two wordings:
I wanted to write "not tracked by autograd here" because it links well with the next sentence (removing tracking implies speed up, which sets us up for the next sentence: "speed up your model even more".)
"behave as if none of the inputs require grad" OTOH is I think still a useful way to think about no-grad mode since its very true to its implementation.
Also simply stating no-grad mode as the mode where there is no gradient tracking seems to obvious anyway.

I'm thinking we can add a sentence in no-grad to link the two interpretations. 
```
Computations in no-grad mode behave as if none of the inputs require grad.
+ In other words, computations under no-grad mode will not be tracked by autograd even if there are inputs that require grad.
```

Reply:
Yes that sounds good. I feel like the "as if no input requires grad" is the clearest for the user.